### PR TITLE
ip allocation lib

### PR DESCRIPTION
**Pull request by henderiw**
library to assist functions to manipulate ip allocations

### CODE REVIEW

**Review by gvbalaji — COMMENTED** [review]
Loos good to me.

**Review by johnbelamaric — CHANGES_REQUESTED** [review]
I think this should be reconsidered along the same lines as my comments in the other library PRs.

File: krm-functions/lib/ipallocation/v1alpha1/ipallocation.go [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+	"github.com/GoogleContainerTools/kpt-functions-sdk/go/fn"
+	ipamv1alpha1 "github.com/nokia/k8s-ipam/apis/ipam/v1alpha1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"sigs.k8s.io/kustomize/kyaml/kio/filters"
+	"sigs.k8s.io/yaml"
+)
+
+
+// IPAllocation interface will refer to all go-functions this library supports
+type IPAllocation interface {
```
Comment (vjayaramrh) [inline krm-functions/lib/ipallocation/v1alpha1/ipallocation.go:29]: Are we going to expect multiple struct definitions to implement the IPAllocation interface? Thanks

File: krm-functions/lib/ipallocation/v1alpha1/ipallocation_test.go [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+package v1alpha1
+
+import (
+	"testing"
+
+	"github.com/nokia/k8s-ipam/apis/ipam/v1alpha1"
+	v1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+func TestParseKubeObjectNonEmpty(t *testing.T) {
```
Comment (vjayaramrh) [inline krm-functions/lib/ipallocation/v1alpha1/ipallocation_test.go:26]: WOuld be a good idea to combine `TestParseKubeObjectNonEmpty` and `TestParseKubeObjectEmpty` into one single test method `TestParseKubeObject` and include both the test cases

File: krm-functions/lib/ipallocation/v1alpha1/ipallocation.go (hunk 1 above)
Comment (henderiw) [reply krm-functions/lib/ipallocation/v1alpha1/ipallocation.go:29]: yes we will align this with the interface one. So we will have NewFromYaml and NewFromGostruct.

File: krm-functions/lib/ipallocation/v1alpha1/ipallocation_test.go (hunk 2 above)
Comment (henderiw) [reply krm-functions/lib/ipallocation/v1alpha1/ipallocation_test.go:26]: yes will do if we refactor this.

**Pull request closed without merging**